WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift

```diff
@@ -89,8 +132,74 @@ public final class UserRepository: UserRepositoryProtocol {
         }
     }
 
+    public func updateUser(
+        from event: UserUpdateEvent
+    ) async throws {
+        try await context.perform { [self] in
+
+            let user = fetchOrCreateUser(
+                with: event.id
+            )
+
+            if isFederationEnabled {
+                user.domain = event.userID.domain
+            }
+
+            if let name = event.name {
+                user.name = name
+            }
+
+            if let email = event.email {
+                user.emailAddress = email
+            }
+
+            if let handle = event.handle {
+                user.handle = handle
+            }
+
+            if let accentColor = event.accentColorID {
+                user.accentColorValue = Int16(accentColor)
+            }
+
+            let assetKeys: Set<String> = [
+                ZMUser.previewProfileAssetIdentifierKey,
+                ZMUser.completeProfileAssetIdentifierKey
+            ]
+
+            /// Do not update assets if user has local modifications.
+            if !user.hasLocalModifications(forKeys: assetKeys) {
+                let previewAssetKey = event.assets?
+                    .first(where: { $0.size == .preview })
+                    .map(\.key)
+
+                let completeAssetKey = event.assets?
+                    .first(where: { $0.size == .complete })
+                    .map(\.key)
+
+                if let previewAssetKey {
+                    user.previewProfileAssetIdentifier = previewAssetKey
+                }
+
+                if let completeAssetKey {
+                    user.completeProfileAssetIdentifier = completeAssetKey
+                }
+            }
+
+            user.supportedProtocols = event.supportedProtocols?.toDomainModel() ?? [.proteus]
+
+            user.isPendingMetadataRefresh = false
+
+            try context.save()
+        }
+    }
+
+    // MARK: - Private
+
     private func persistUser(from user: WireAPI.User) {
```

Comment:
There seems to be some overlap with this method and the `updateUser` method. Maybe not now, but do you think it's worth at some point to find a solution that minimizes this duplication? I suppose it could be achieved with an intermediate `UpdateInfo` struct. Maybe something to consider if we introduce a proper data layer.

Reply:
Yes, I did some refactoring to avoid code duplication in this [here](https://github.com/wireapp/wire-ios/pull/1999) using an intermediate `updateUserMetadata(..)` func. So I'll be able to reuse it. 
I'll just mark a comment for now on this one.

Reply:
done in https://github.com/wireapp/wire-ios/pull/1958/commits/411d1ab958b6f0aa0735b39903db173c84783820